go/datas/http_batch_store.go

```diff
@@ -313,16 +332,21 @@ func resBodyReader(res *http.Response) (reader io.ReadCloser) {
 }
 
 func (bhcs *httpBatchStore) SchedulePut(c chunks.Chunk, refHeight uint64, hints types.Hints) {
-	if !bhcs.unwrittenPuts.Insert(c, refHeight) {
+	tryCache := func(c chunks.Chunk) bool {
+		bhcs.cacheMu.RLock()
+		defer bhcs.cacheMu.RUnlock()
+		return bhcs.unwrittenPuts.Insert(c)
+	}
+	if !tryCache(c) {
 		return
 	}
 
 	bhcs.requestWg.Add(1)
-	bhcs.writeQueue <- writeRequest{c.Hash(), hints, false}
+	bhcs.writeQueue <- writeRequest{hints, false}
```

Comment:
I'm wondering whether the machinery around the write queue (the channel, the persistent goroutine that are servicing the write-queue and the flush queue) are just unneeded complexity now. this stuff is a hold over from the before time when we didn't have any validation and thus were writing in larger batches, but we haven't done that in a while.

i think it's possible now to just have `SchedulePut` Put into the nbs (not care about whether the chunk was contained), and add its hints to the hint set, all inside the mutex, and have `Flush` just send everything over.

i'm probably forgetting about some critical thing, right?